scwx-qt/source/scwx/qt/ui/marker_settings_widget.cpp

```diff
@@ -65,21 +65,25 @@ void MarkerSettingsWidgetImpl::ConnectSignals()
                     {
                        markerManager_->add_marker(types::MarkerInfo("", 0, 0));
                     });
-   QObject::connect(self_->ui->removeButton,
-                    &QPushButton::clicked,
-                    self_,
-                    [this]()
-                    {
-                       auto selectionModel =
-                          self_->ui->markerView->selectionModel();
-                       QModelIndex selected =
-                          selectionModel
-                             ->selectedRows(static_cast<int>(
-                                model::MarkerModel::Column::Name))
-                             .first();
-
-                       markerManager_->remove_marker(selected.row());
-                    });
+   QObject::connect(
+      self_->ui->removeButton,
+      &QPushButton::clicked,
+      self_,
+      [this]()
+      {
+         auto        selectionModel = self_->ui->markerView->selectionModel();
+         QModelIndex selected       = selectionModel
+                                   ->selectedRows(static_cast<int>(
+                                      model::MarkerModel::Column::Name))
+                                   .first();
+         std::optional<types::MarkerId> id = markerModel_->getId(selected.row());
```

Comment:
Should this reference `p->markerIds_` instead?

Reply:
Can you double check this comment. There is no `markerIds_` in this file, and getting the id from `markerModel_` seams correct to me (`markerModel_->getId` correctly uses `markerIds_`).

Reply:
You're correct, rescinding comment.